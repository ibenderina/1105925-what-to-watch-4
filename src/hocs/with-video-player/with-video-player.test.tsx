import * as React from "react";
import * as renderer from "react-test-renderer";
import withVideoPlayer from "@hocs/with-video-player/with-video-player";

interface MockComponentProps {
  children: React.ReactNode;
}

const MockComponent = (props: MockComponentProps) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

const MockComponentWrapped = withVideoPlayer(MockComponent);

it(`withVideoPlayer is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped image={'/'} url={'/'}>Test</MockComponentWrapped>
  ), {
    createNodeMock() {
      return {};
    }
  });

  expect(tree.toJSON()).toMatchSnapshot();
});
