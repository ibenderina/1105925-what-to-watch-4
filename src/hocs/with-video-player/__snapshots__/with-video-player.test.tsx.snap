// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withVideoPlayer is rendered correctly 1`] = `
<div>
  <video
    autoPlay={true}
    height="175"
    loop={true}
    muted={true}
    poster="/"
    src=""
    width="280"
  />
</div>
`;
